src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefLocalTests.cs

```diff
@@ -12,6 +12,253 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
     [CompilerTrait(CompilerFeature.RefLocalsReturns)]
     public class CodeGenRefLocalTests : CompilingTestBase
     {
+        [Fact]
+        public void RefReassignInArrayElement()
+        {
+            const string src = @"
+using System;
+class C
+{
+    void M()
+    {
+        object o = string.Empty;
+        M2(o);
+    }
+    void M2(in object o)
+    {
+        o = ref (new object[1])[0];
+        Console.WriteLine(o?.GetHashCode() ?? 5);
+    }
+}";
+            var verifier = CompileAndVerify(src, verify: Verification.Fails);
+            const string expectedIL = @"
+{
+  // Code size       36 (0x24)
+  .maxstack  2
+  IL_0000:  ldc.i4.1
+  IL_0001:  newarr     ""object""
+  IL_0006:  ldc.i4.0
+  IL_0007:  readonly.
+  IL_0009:  ldelema    ""object""
+  IL_000e:  starg.s    V_1
+  IL_0010:  ldarg.1
+  IL_0011:  ldind.ref
+  IL_0012:  dup
+  IL_0013:  brtrue.s   IL_0019
+  IL_0015:  pop
+  IL_0016:  ldc.i4.5
+  IL_0017:  br.s       IL_001e
+  IL_0019:  callvirt   ""int object.GetHashCode()""
+  IL_001e:  call       ""void System.Console.WriteLine(int)""
+  IL_0023:  ret
+}";
+            verifier.VerifyIL("C.M2", expectedIL);
+
+            // N.B. Even with PEVerify compat this generates unverifiable code.
```

Comment:
Right. We do not respect PEVerify-compat when that would result in functional break of the feature. 

Another example is ref-returning a readonly field. PEVerify compat flag generally makes us to do local copies before taking a ref to a readonly field, but returning a ref to a local clone makes no sense. Having a choice between not compatible and completely broken we pick not compatible.